src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -326,47 +361,29 @@ void ExecutorWorker::handleCreateOp(SyncOpPtr syncOp, std::shared_ptr<AbstractJo
                     syncOp->affectedNode()->lastmodified(), node)) {
             LOGW_SYNCPAL_WARN(_logger, L"Failed to propagate changes in DB or update tree for: "
                                                << SyncName2WStr(syncOp->affectedNode()->name()).c_str());
-            _syncPal->setProgressComplete(relativeLocalFilePath, SyncFileStatus::Error);
+            // _executorExitCode and _executorExitCause are set by the above function
             hasError = true;
             return;
         }
-
-        _syncPal->setProgressComplete(relativeLocalFilePath, SyncFileStatus::Success);
     } else {
         if (!isLiteSyncActivated() && !enoughLocalSpace(syncOp)) {
+            _syncPal->addError(Error(_syncPal->syncDbId(), name(), _executorExitCode, _executorExitCause));
```

Comment:
So now we are creating the error before setting the ExitCode and ExitCause? How do we know that this error is about a disk full issue?

Reply:
It's an issue. Fixed.